### ISSUE TITLE
pipenv 2023.6.26 breaks installer

### ISSUE THREAD

**Issue by txoof** [txoof/PaperPi#108]
**Describe the bug**
pipenv version 2023.6.26 appears to fail with installing paperpi.

**To Reproduce**
1. update to pipenv 2023.6.26
2. run installer 
3. package installation will fail (see below)

```
he flag --skip-lock has been deprecated for removal.  Without running the lock resolver it is not possible to manage multiple package indexes.  Additionally it bypasses the build consistency guarantees provided by maintaining a lock file.
Requirements file provided! Importing into Pipfile...
Installing dependencies from Pipfile...
INFO:root:running egg_info
INFO:root:creating /tmp/reqlib-srci_y9s8dv/it8951/reqlib-metadata/IT8951.egg-info
INFO:root:writing /tmp/reqlib-srci_y9s8dv/it8951/reqlib-metadata/IT8951.egg-info/PKG-INFO
INFO:root:writing dependency_links to /tmp/reqlib-srci_y9s8dv/it8951/reqlib-metadata/IT8951.egg-info/dependency_links.txt
INFO:root:writing top-level names to /tmp/reqlib-srci_y9s8dv/it8951/reqlib-metadata/IT8951.egg-info/top_level.txt
INFO:root:writing manifest file '/tmp/reqlib-srci_y9s8dv/it8951/reqlib-metadata/IT8951.egg-info/SOURCES.txt'
INFO:root:reading manifest file '/tmp/reqlib-srci_y9s8dv/it8951/reqlib-metadata/IT8951.egg-info/SOURCES.txt'
INFO:root:adding license file 'LICENSE'
INFO:root:writing manifest file '/tmp/reqlib-srci_y9s8dv/it8951/reqlib-metadata/IT8951.egg-info/SOURCES.txt'
An error occurred while installing dictor! Will try again.
An error occurred while installing pytz! Will try again.
An error occurred while installing feedparser! Will try again.
An error occurred while installing currency-symbols! Will try again.
An error occurred while installing pycoingecko! Will try again.
An error occurred while installing pygal! Will try again.
An error occurred while installing qrcode! Will try again.
An error occurred while installing cairocffi! Will try again.
An error occurred while installing cffi! Will try again.
An error occurred while installing cairosvg! Will try again.
An error occurred while installing psutil! Will try again.
An error occurred while installing python-dateutil! Will try again.
An error occurred while installing querylms! Will try again.
Installing initially failed dependencies...
[pipenv.exceptions.InstallError]: Looking in indexes: https://pypi.org/simple, https://www.piwheels.org/simple
[pipenv.exceptions.InstallError]: Collecting dictor (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 1))
[pipenv.exceptions.InstallError]:   Using cached dictor-0.1.11-py3-none-any.whl
[pipenv.exceptions.InstallError]: Collecting pytz (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 2))
[pipenv.exceptions.InstallError]:   Using cached pytz-2023.3-py2.py3-none-any.whl (502 kB)
[pipenv.exceptions.InstallError]: Collecting feedparser (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 3))
[pipenv.exceptions.InstallError]:   Using cached feedparser-6.0.10-py3-none-any.whl (81 kB)
[pipenv.exceptions.InstallError]: Collecting currency-symbols (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 4))
[pipenv.exceptions.InstallError]:   Using cached currency_symbols-2.0.3-py3-none-any.whl (5.0 kB)
[pipenv.exceptions.InstallError]: Collecting pycoingecko (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 5))
[pipenv.exceptions.InstallError]:   Using cached pycoingecko-3.1.0-py3-none-any.whl (8.8 kB)
[pipenv.exceptions.InstallError]: Collecting pygal (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 6))
[pipenv.exceptions.InstallError]:   Using cached pygal-3.0.0-py2.py3-none-any.whl (129 kB)
[pipenv.exceptions.InstallError]: Collecting qrcode (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 7))
[pipenv.exceptions.InstallError]:   Using cached qrcode-7.4.2-py3-none-any.whl (46 kB)
[pipenv.exceptions.InstallError]: Collecting cairocffi (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 8))
[pipenv.exceptions.InstallError]:   Using cached cairocffi-1.6.0-py3-none-any.whl (75 kB)
[pipenv.exceptions.InstallError]: Collecting cffi (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 9))
[pipenv.exceptions.InstallError]:   Using cached cffi-1.15.1.tar.gz (508 kB)
[pipenv.exceptions.InstallError]:   Preparing metadata (setup.py): started
[pipenv.exceptions.InstallError]:   Preparing metadata (setup.py): finished with status 'done'
[pipenv.exceptions.InstallError]: Collecting cairosvg (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 10))
[pipenv.exceptions.InstallError]:   Using cached CairoSVG-2.7.0-py3-none-any.whl (43 kB)
[pipenv.exceptions.InstallError]: Collecting psutil (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 11))
[pipenv.exceptions.InstallError]:   Using cached psutil-5.9.5.tar.gz (493 kB)
[pipenv.exceptions.InstallError]:   Installing build dependencies: started
[pipenv.exceptions.InstallError]:   Installing build dependencies: finished with status 'done'
[pipenv.exceptions.InstallError]:   Getting requirements to build wheel: started
[pipenv.exceptions.InstallError]:   Getting requirements to build wheel: finished with status 'done'
[pipenv.exceptions.InstallError]:   Preparing metadata (pyproject.toml): started
[pipenv.exceptions.InstallError]:   Preparing metadata (pyproject.toml): finished with status 'done'
[pipenv.exceptions.InstallError]: Collecting python-dateutil (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 12))
[pipenv.exceptions.InstallError]:   Using cached python_dateutil-2.8.2-py2.py3-none-any.whl (247 kB)
[pipenv.exceptions.InstallError]: Collecting querylms (from -r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 13))
[pipenv.exceptions.InstallError]:   Using cached QueryLMS-0.2.2.0-py3-none-any.whl (11 kB)
[pipenv.exceptions.InstallError]: Collecting sgmllib3k (from feedparser->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 3))
[pipenv.exceptions.InstallError]:   Using cached sgmllib3k-1.0.0-py3-none-any.whl
[pipenv.exceptions.InstallError]: Requirement already satisfied: requests in ./.venv/lib/python3.9/site-packages (from pycoingecko->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 5)) (2.31.0)
[pipenv.exceptions.InstallError]: Collecting typing-extensions (from qrcode->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 7))
[pipenv.exceptions.InstallError]:   Using cached typing_extensions-4.6.3-py3-none-any.whl (31 kB)
[pipenv.exceptions.InstallError]: Collecting pypng (from qrcode->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 7))
[pipenv.exceptions.InstallError]:   Using cached pypng-0.20220715.0-py3-none-any.whl (58 kB)
[pipenv.exceptions.InstallError]: Collecting pycparser (from cffi->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 9))
[pipenv.exceptions.InstallError]:   Using cached pycparser-2.21-py2.py3-none-any.whl (118 kB)
[pipenv.exceptions.InstallError]: Collecting cssselect2 (from cairosvg->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 10))
[pipenv.exceptions.InstallError]:   Using cached cssselect2-0.7.0-py3-none-any.whl (15 kB)
[pipenv.exceptions.InstallError]: Collecting defusedxml (from cairosvg->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 10))
[pipenv.exceptions.InstallError]:   Using cached defusedxml-0.7.1-py2.py3-none-any.whl (25 kB)
[pipenv.exceptions.InstallError]: Requirement already satisfied: pillow in ./.venv/lib/python3.9/site-packages (from cairosvg->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 10)) (9.5.0)
[pipenv.exceptions.InstallError]: Collecting tinycss2 (from cairosvg->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 10))
[pipenv.exceptions.InstallError]:   Using cached tinycss2-1.2.1-py3-none-any.whl (21 kB)
[pipenv.exceptions.InstallError]: Collecting six>=1.5 (from python-dateutil->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 12))
[pipenv.exceptions.InstallError]:   Using cached six-1.16.0-py2.py3-none-any.whl (11 kB)
[pipenv.exceptions.InstallError]: Collecting webencodings (from cssselect2->cairosvg->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 10))
[pipenv.exceptions.InstallError]:   Using cached webencodings-0.5.1-py2.py3-none-any.whl (11 kB)
[pipenv.exceptions.InstallError]: Requirement already satisfied: charset-normalizer<4,>=2 in ./.venv/lib/python3.9/site-packages (from requests->pycoingecko->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 5)) (3.1.0)
[pipenv.exceptions.InstallError]: Requirement already satisfied: idna<4,>=2.5 in ./.venv/lib/python3.9/site-packages (from requests->pycoingecko->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 5)) (3.4)
[pipenv.exceptions.InstallError]: Requirement already satisfied: urllib3<3,>=1.21.1 in ./.venv/lib/python3.9/site-packages (from requests->pycoingecko->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 5)) (2.0.3)
[pipenv.exceptions.InstallError]: Requirement already satisfied: certifi>=2017.4.17 in ./.venv/lib/python3.9/site-packages (from requests->pycoingecko->-r /tmp/pipenv-nia7mhqz-requirements/pipenv-ge9hs4lw-hashed-reqs.txt (line 5)) (2023.5.7)
[pipenv.exceptions.InstallError]: Building wheels for collected packages: cffi, psutil
[pipenv.exceptions.InstallError]:   Building wheel for cffi (setup.py): started
[pipenv.exceptions.InstallError]:   Building wheel for cffi (setup.py): finished with status 'error'
[pipenv.exceptions.InstallError]:   Running setup.py clean for cffi
[pipenv.exceptions.InstallError]:   Building wheel for psutil (pyproject.toml): started
[pipenv.exceptions.InstallError]:   Building wheel for psutil (pyproject.toml): finished with status 'done'
[pipenv.exceptions.InstallError]:   Created wheel for psutil: filename=psutil-5.9.5-cp39-abi3-linux_aarch64.whl size=276408 sha256=352e2554456c1d7289b4054c6b37b69b016628630c022b6b01fa5c7fa4da30ca
[pipenv.exceptions.InstallError]:   Stored in directory: /root/.cache/pipenv/wheels/74/d1/7d/d9ae7d9aea0f1cebed73f37868df7b5f3333e7f30163b3e558
[pipenv.exceptions.InstallError]: Successfully built psutil
[pipenv.exceptions.InstallError]: Failed to build cffi
[pipenv.exceptions.InstallError]: error: subprocess-exited-with-error
[pipenv.exceptions.InstallError]:
[pipenv.exceptions.InstallError]:   × python setup.py bdist_wheel did not run successfully.
[pipenv.exceptions.InstallError]:   │ exit code: 1
[pipenv.exceptions.InstallError]:   ╰─> [72 lines of output]
[pipenv.exceptions.InstallError]:       Package libffi was not found in the pkg-config search path.
[pipenv.exceptions.InstallError]:       Perhaps you should add the directory containing `libffi.pc'
[pipenv.exceptions.InstallError]:       to the PKG_CONFIG_PATH environment variable
[pipenv.exceptions.InstallError]:       No package 'libffi' found
[pipenv.exceptions.InstallError]:       Package libffi was not found in the pkg-config search path.
[pipenv.exceptions.InstallError]:       Perhaps you should add the directory containing `libffi.pc'
[pipenv.exceptions.InstallError]:       to the PKG_CONFIG_PATH environment variable
[pipenv.exceptions.InstallError]:       No package 'libffi' found
[pipenv.exceptions.InstallError]:       Package libffi was not found in the pkg-config search path.
[pipenv.exceptions.InstallError]:       Perhaps you should add the directory containing `libffi.pc'
[pipenv.exceptions.InstallError]:       to the PKG_CONFIG_PATH environment variable
[pipenv.exceptions.InstallError]:       No package 'libffi' found
[pipenv.exceptions.InstallError]:       Package libffi was not found in the pkg-config search path.
[pipenv.exceptions.InstallError]:       Perhaps you should add the directory containing `libffi.pc'
[pipenv.exceptions.InstallError]:       to the PKG_CONFIG_PATH environment variable
[pipenv.exceptions.InstallError]:       No package 'libffi' found
[pipenv.exceptions.InstallError]:       Package libffi was not found in the pkg-config search path.
[pipenv.exceptions.InstallError]:       Perhaps you should add the directory containing `libffi.pc'
[pipenv.exceptions.InstallError]:       to the PKG_CONFIG_PATH environment variable
[pipenv.exceptions.InstallError]:       No package 'libffi' found
[pipenv.exceptions.InstallError]:       /usr/local/paperpi/.venv/lib/python3.9/site-packages/setuptools/config/setupcfg.py:293: _DeprecatedConfig: Deprecated config in `setup.cfg`
[pipenv.exceptions.InstallError]:       !!
[pipenv.exceptions.InstallError]:
[pipenv.exceptions.InstallError]:               ********************************************************************************
[pipenv.exceptions.InstallError]:               The license_file parameter is deprecated, use license_files instead.
[pipenv.exceptions.InstallError]:
[pipenv.exceptions.InstallError]:               By 2023-Oct-30, you need to update your project and remove deprecated calls
[pipenv.exceptions.InstallError]:               or your builds will no longer be supported.
[pipenv.exceptions.InstallError]:
[pipenv.exceptions.InstallError]:               See https://setuptools.pypa.io/en/latest/userguide/declarative_config.html for details.
[pipenv.exceptions.InstallError]:               ********************************************************************************
[pipenv.exceptions.InstallError]:
[pipenv.exceptions.InstallError]:       !!
[pipenv.exceptions.InstallError]:         parsed = self.parsers.get(option_name, lambda x: x)(value)
[pipenv.exceptions.InstallError]:       running bdist_wheel
[pipenv.exceptions.InstallError]:       running build
[pipenv.exceptions.InstallError]:       running build_py
[pipenv.exceptions.InstallError]:       creating build
[pipenv.exceptions.InstallError]:       creating build/lib.linux-aarch64-cpython-39
[pipenv.exceptions.InstallError]:       creating build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/error.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/__init__.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/commontypes.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/lock.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/pkgconfig.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/setuptools_ext.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/ffiplatform.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/backend_ctypes.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/vengine_gen.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/model.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/api.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/verifier.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/cffi_opcode.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/recompiler.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/vengine_cpy.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/cparser.py -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/_cffi_include.h -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/parse_c_type.h -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/_embedding.h -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       copying cffi/_cffi_errors.h -> build/lib.linux-aarch64-cpython-39/cffi
[pipenv.exceptions.InstallError]:       warning: build_py: byte-compiling is disabled, skipping.
[pipenv.exceptions.InstallError]:
[pipenv.exceptions.InstallError]:       running build_ext
[pipenv.exceptions.InstallError]:       building '_cffi_backend' extension
[pipenv.exceptions.InstallError]:       creating build/temp.linux-aarch64-cpython-39
[pipenv.exceptions.InstallError]:       creating build/temp.linux-aarch64-cpython-39/c
[pipenv.exceptions.InstallError]:       arm-linux-gnueabihf-gcc -pthread -Wno-unused-result -Wsign-compare -DNDEBUG -g -fwrapv -O2 -Wall -g -ffile-prefix-map=/python3.9-3.9.2=. -fstack-protector-strong -Wformat -Werror=format-security -g -fwrapv -O2 -fPIC -DFFI_BUILDING=1 -DUSE__THREAD -DHAVE_SYNC_SYNCHRONIZE -I/usr/include/ffi -I/usr/include/libffi -I/usr/local/paperpi/.venv/include -I/usr/include/python3.9 -c c/_cffi_backend.c -o build/temp.linux-aarch64-cpython-39/c/_cffi_backend.o
[pipenv.exceptions.InstallError]:       c/_cffi_backend.c:15:10: fatal error: ffi.h: No such file or directory
[pipenv.exceptions.InstallError]:          15 | #include <ffi.h>
[pipenv.exceptions.InstallError]:             |          ^~~~~~~
[pipenv.exceptions.InstallError]:       compilation terminated.
[pipenv.exceptions.InstallError]:       error: command '/usr/bin/arm-linux-gnueabihf-gcc' failed with exit code 1
[pipenv.exceptions.InstallError]:       [end of output]
[pipenv.exceptions.InstallError]:
[pipenv.exceptions.InstallError]:   note: This error originates from a subprocess, and is likely not a problem with pip.
[pipenv.exceptions.InstallError]:   ERROR: Failed building wheel for cffi
[pipenv.exceptions.InstallError]: ERROR: Could not build wheels for cffi, which is required to install pyproject.toml-based projects
ERROR: Couldn't install package: [Requirement(vcs=None, req=NamedRequirement(name='dictor', version='', req=Requirement.parse('dictor'), extras=[], editable=False, parsed_line=<Line (editable=False, name=dictor, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='dictor'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='dictor', normalized_name='dictor', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=dictor, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=dictor)>), Requirement(vcs=None, req=NamedRequirement(name='pytz', version='', req=Requirement.parse('pytz'), extras=[], editable=False, parsed_line=<Line (editable=False, name=pytz, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='pytz'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='pytz', normalized_name='pytz', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=pytz, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=pytz)>), Requirement(vcs=None, req=NamedRequirement(name='feedparser', version='', req=Requirement.parse('feedparser'), extras=[], editable=False, parsed_line=<Line (editable=False, name=feedparser, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='feedparser'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='feedparser', normalized_name='feedparser', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=feedparser, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=feedparser)>), Requirement(vcs=None, req=NamedRequirement(name='currency-symbols', version='', req=Requirement.parse('currency-symbols'), extras=[], editable=False, parsed_line=<Line (editable=False, name=currency-symbols, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='currency-symbols'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='currency-symbols', normalized_name='currency-symbols', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=currency-symbols, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=currency-symbols)>), Requirement(vcs=None, req=NamedRequirement(name='pycoingecko', version='', req=Requirement.parse('pycoingecko'), extras=[], editable=False, parsed_line=<Line (editable=False, name=pycoingecko, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='pycoingecko'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='pycoingecko', normalized_name='pycoingecko', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=pycoingecko, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=pycoingecko)>), Requirement(vcs=None, req=NamedRequirement(name='pygal', version='', req=Requirement.parse('pygal'), extras=[], editable=False, parsed_line=<Line (editable=False, name=pygal, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='pygal'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='pygal', normalized_name='pygal', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=pygal, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=pygal)>), Requirement(vcs=None, req=NamedRequirement(name='qrcode', version='', req=Requirement.parse('qrcode'), extras=[], editable=False, parsed_line=<Line (editable=False, name=qrcode, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='qrcode'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='qrcode', normalized_name='qrcode', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=qrcode, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=qrcode)>), Requirement(vcs=None, req=NamedRequirement(name='cairocffi', version='', req=Requirement.parse('cairocffi'), extras=[], editable=False, parsed_line=<Line (editable=False, name=cairocffi, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='cairocffi'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='cairocffi', normalized_name='cairocffi', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=cairocffi, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=cairocffi)>), Requirement(vcs=None, req=NamedRequirement(name='cffi', version='', req=Requirement.parse('cffi'), extras=[], editable=False, parsed_line=<Line (editable=False, name=cffi, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='cffi'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='cffi', normalized_name='cffi', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=cffi, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=cffi)>), Requirement(vcs=None, req=NamedRequirement(name='cairosvg', version='', req=Requirement.parse('cairosvg'), extras=[], editable=False, parsed_line=<Line (editable=False, name=cairosvg, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='cairosvg'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='cairosvg', normalized_name='cairosvg', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=cairosvg, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=cairosvg)>), Requirement(vcs=None, req=NamedRequirement(name='psutil', version='', req=Requirement.parse('psutil'), extras=[], editable=False, parsed_line=<Line (editable=False, name=psutil, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='psutil'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='psutil', normalized_name='psutil', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=psutil, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=psutil)>), Requirement(vcs=None, req=NamedRequirement(name='python-dateutil', version='', req=Requirement.parse('python-dateutil'), extras=[], editable=False, parsed_line=<Line (editable=False, name=python-dateutil, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='python-dateutil'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='python-dateutil', normalized_name='python-dateutil', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=python-dateutil, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=python-dateutil)>), Requirement(vcs=None, req=NamedRequirement(name='querylms', version='', req=Requirement.parse('querylms'), extras=[], editable=False, parsed_line=<Line (editable=False, name=querylms, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=None)>, line_part='querylms'), markers=None, index=None, editable=False, hashes=set(), extras=(), name='querylms', normalized_name='querylms', is_file_or_url=False, is_vcs=False, extras_as_pip='', get_specifiers='', specifiers=None, hashes_as_pip='', _line_instance=<Line (editable=False, name=querylms, path=None, uri=None, extras=(), markers=None, vcs=None, specifier=None, pyproject=None, pyproject_requires=None, pyproject_backend=None, ireq=querylms)>)]
 Package installation failed...
/home/pi
failed to install python modules
```

**Expected behavior**
Installer should create a pipenv and install necessary packages

**Additional context**
This may be related to the reliance on `--skip-lock`

**Comment by txoof** [comment]
Looks like this may be related to PIL/Pillow missing zlib -- there doesn't appear to be any good way to get zlib linked so Pillow can be built by pip. The solution may be to switch to `--site-packages` and install the native debian package of Pillow.

**Comment by txoof** [comment]
This problem is due to changes in how [pipenv manages pypi indexes](https://github.com/pypa/pipenv/issues/5756). 

The simple solution looks to be to change the Pipfile so points to ` URL = " https://www.piwheels.org/simple/"`

**Comment by txoof** [comment]
resolved with #109
